### PR TITLE
The Final* Foodcode Fix

### LINKED ISSUE #5195: Korlaaskak cannot be ate
It's apparently slicable, yet when you slice it, nothing happens. It just keeps saying "You sliced the Korlaaska!" and it doesn't actually divide into portions. 

You cannot eat the Korlaaska unsliced. 

You can "poke it with a fork" and the sprite of food shows up on said fork, however, when you go to eat it, it says nothing is on the fork.

The Korlaaska also has a set transfer amount, but that doesn't do anything either.

### LINKED ISSUE #5205: Eating food is not respecting bitesize
As the title says, eating food seems to always transfer five units, instead of whatever the bitesize is set.

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
Foodcode was improved. Previously it was fucking garbage for using snowflake code but now it works as it should.

Fixes #5205
Fixes #5195
Makes it so that nachos have a better system of picking up a nacho plate.

### CODE REVIEW

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -60,108 +60,85 @@
...
+			var/mob/living/carbon/human/H = target
+			var/limit = 550 * (1 + H.overeatduration / 2000)
+			var/current = H.nutrition + (H.reagents.get_reagent_amount("nutriment") * 25)
+			is_full = current >= limit
+
+		if(user == target)
+			if(!user.can_eat(src))
+				return
+			if(is_full)
+				to_chat(user,span("warning","You  can't stomach any more food!"))
```
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: im a failure

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -60,108 +60,85 @@
...
-	return 0
+			var/contained = reagentlist()
+			target.attack_log += text("\[[time_stamp()]\] <font color='orange'>Has been fed [name] by [key_name(user)] Reagents: [contained]</font>")
+			user.attack_log += text("\[[time_stamp()]\] <font color='red'>Fed [name] to [key_name(target)] Reagents: [contained]</font>")
+			msg_admin_attack("[key_name_admin(user)] fed [key_name_admin(target)] with [name] Reagents: [contained] (INTENT: [uppertext(user.a_intent)]) (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)",ckey=key_name(user),ckey_target=key_name(target))
+			reagents.trans_to_mob(target, min(reagents.total_volume,bitesize), CHEM_INGEST)
+
+	feed_sound(user)
+	bitecount++
+	On_Consume(target,user)
```
Comment (MarinaGryphon) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Is this a new proc? If so, it should probably be `on_consume(target,user)` to match other procs. Snake case, all lowercase, or camel case.

Comment (BurgerLUA) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: It's an old proc.

**Comment by BurgerLUA** [comment]
travis canceled for some reason and needs a reboot

**Pull request merged**